File: two_sum#1/solution_1.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
  const hash = [];
  for (let [index, value] of nums.entries()) {
    if (Number.isInteger(hash[value])) {
      return [hash[value], index];
    }
    hash[target-value] = index;
  }
  return [];
};

File: two_sum#1/solution_1.rb
# @param {Integer[]} nums
# @param {Integer} target
# @return {Integer[]}
def two_sum(nums, target)
  res, hash = nil, {}
  nums.each_with_index do |num, index|
    res ||= [hash[num], index] if hash[num]
    hash[target-num] = index
  end
  res
end
